node/forest_libp2p/src/service.rs

```diff
@@ -104,6 +105,19 @@ pub enum NetworkMessage {
         cid: Cid,
         response_channel: OneShotSender<()>,
     },
+    JSONRPCRequest {
+        method: NetRPCMethods,
+        response_channel: OneShotSender<NetRPCResponse>,
```

Comment:
why not have the channel type based on the request enum? That way all responses don't have to be wrapped in an enum and be matched on when you know what the return type should be?